src/settings-ui/Settings.UI.Library/ViewModels/GeneralViewModel.cs

```diff
@@ -41,11 +63,18 @@ public class GeneralViewModel : Observable
 
         private IFileSystemWatcher _fileWatcher;
 
-        public GeneralViewModel(ISettingsRepository<GeneralSettings> settingsRepository, string runAsAdminText, string runAsUserText, bool isElevated, bool isAdmin, Func<string, int> updateTheme, Func<string, int> ipcMSGCallBackFunc, Func<string, int> ipcMSGRestartAsAdminMSGCallBackFunc, Func<string, int> ipcMSGCheckForUpdatesCallBackFunc, string configFileSubfolder = "", Action dispatcherAction = null)
+        public GeneralViewModel(ISettingsRepository<GeneralSettings> settingsRepository, string runAsAdminText, string runAsUserText, bool isElevated, bool isAdmin, Func<string, int> updateTheme, Func<string, int> ipcMSGCallBackFunc, Func<string, int> ipcMSGRestartAsAdminMSGCallBackFunc, Func<string, int> ipcMSGCheckForUpdatesCallBackFunc, string configFileSubfolder = "", Action dispatcherAction = null, Action hideBackupAndRestoreMessageAreaAction = null, object resourceLoader = null)
         {
             CheckForUpdatesEventHandler = new ButtonClickCommand(CheckForUpdatesClick);
             RestartElevatedButtonEventHandler = new ButtonClickCommand(RestartElevated);
             UpdateNowButtonEventHandler = new ButtonClickCommand(UpdateNowClick);
+            BackupConfigsEventHandler = new ButtonClickCommand(BackupConfigsClick);
+            SelectSettingBackupDirEventHandler = new ButtonClickCommand(SelectSettingBackupDir);
+            RestoreConfigsEventHandler = new ButtonClickCommand(RestoreConfigsClick);
+            RestartButtonEventHandler = new ButtonClickCommand(Restart);
+            HideBackupAndRestoreMessageAreaAction = hideBackupAndRestoreMessageAreaAction;
+
+            ResourceLoader = resourceLoader;
```

Comment:
I don't know. It does seem silly because it seems like it should not be needed. 

However, I don't know how to obtain a reference to Resources here. Do you know or know someone to ask?